data_validation/__main__.py

```diff
@@ -314,18 +312,23 @@ def build_config_from_args(args: Namespace, config_manager: ConfigManager):
             _get_calculated_config(args, config_manager)
         )
 
-        # Append Comparison fields
-        if args.comparison_fields:
-            config_manager.append_comparison_fields(
-                _get_comparison_config(args, config_manager)
-            )
-
         # Append primary_keys
         primary_keys = cli_tools.get_arg_list(args.primary_keys)
+        if not primary_keys:
+            primary_keys = config_manager.default_primary_keys()
```

Comment:
Maybe we could put the `config_manager.default_primary_keys()` method calling directly in line 316 in a lambda format? Because I'm not sure about having duplicate `if not primary_keys:` blocks in sequence and with that change we'd only have the exception handling one

Reply:
I don't quite understand. This is the current code:
```
316        primary_keys = cli_tools.get_arg_list(args.primary_keys)
317        if not primary_keys:
318            primary_keys = config_manager.auto_list_primary_keys()
319        if not primary_keys:
320            raise ValueError("--primary-keys argument is required for this validation")
```

- We set primary keys to user input (L316).
- If the user didn't give us input (L317) we set primary keys from the source or target (L318).
- If we didn't get primary keys from source/target (L319) we fail the execution

It could be condensed to below if you think that is less confusing?
```
316        primary_keys = cli_tools.get_arg_list(args.primary_keys) or config_manager.auto_list_primary_keys()
317        if not primary_keys:
318            raise ValueError("--primary-keys argument is required for this validation")
```

Reply:
Yes, I was thinking about the condensed version you put! Not necessarily because it's confusing right now, I just thought having both methods in one line would be more straightforward than having two identical `if` blocks one after the other. But at the end it's up to you, both approaches work and are understandable!

Reply:
Understood. I've implemented the condensed version. Thanks

Reply:
Line 324 (old line #316) - is strange!! - Why not use `args.primary_keys`, just like in the old line # 318 `args.comparison_fields`. Since primary_keys is now optional, we should have a default value of None - see comment below.